tests/test_openjdk.py

```diff
@@ -67,28 +67,38 @@ def test_jdk_version(container, java_version):
     indirect=["container_per_test"],
 )
 @pytest.mark.parametrize(
-    "test_to_run, expected_strings, expected_exit_status, java_params",
+    "test_to_run, expected_strings, expected_exit_status, java_params, arguments, environment",
```

Comment:
I don't know if it is possible, but I would separate the arguments, environments tests into different functions. Here we have one mighty "it does everything" function, which decreases the visibility on the test failures.

Reply:
They are appearing as separate test runs, see the `openjdk` test runs:

```
tests/test_openjdk.py::test_jdk_extended[local-threads_concurrency_and_sleep-expected_strings0-expected_exit_status0----bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 15%]
tests/test_openjdk.py::test_jdk_extended[local-threads_concurrency_and_sleep-expected_strings0-expected_exit_status0----bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 20%]
tests/test_openjdk.py::test_jdk_extended[local-time-expected_strings1-expected_exit_status1----bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 25%]
tests/test_openjdk.py::test_jdk_extended[local-time-expected_strings1-expected_exit_status1----bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 30%]
tests/test_openjdk.py::test_jdk_extended[local-files_and_directories-expected_strings2-expected_exit_status2----bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 35%]
tests/test_openjdk.py::test_jdk_extended[local-files_and_directories-expected_strings2-expected_exit_status2----bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 40%]
tests/test_openjdk.py::test_jdk_extended[local-memory-expected_strings3-expected_exit_status3--Xmx10M---bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 45%]
tests/test_openjdk.py::test_jdk_extended[local-memory-expected_strings3-expected_exit_status3--Xmx10M---bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 50%]
tests/test_openjdk.py::test_jdk_extended[local-garbage_collector-expected_strings4-expected_exit_status4----bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 55%]
tests/test_openjdk.py::test_jdk_extended[local-garbage_collector-expected_strings4-expected_exit_status4----bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 60%]
tests/test_openjdk.py::test_jdk_extended[local-system_exit-expected_strings5-expected_exit_status5--2--bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 65%]
tests/test_openjdk.py::test_jdk_extended[local-system_exit-expected_strings5-expected_exit_status5--2--bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 70%]
tests/test_openjdk.py::test_jdk_extended[local-system_out-expected_strings6-expected_exit_status6----bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 75%]
tests/test_openjdk.py::test_jdk_extended[local-system_out-expected_strings6-expected_exit_status6----bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 80%]
tests/test_openjdk.py::test_jdk_extended[local-system_properties-expected_strings7-expected_exit_status7--Dproperty1=test---bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 85%]
tests/test_openjdk.py::test_jdk_extended[local-system_properties-expected_strings7-expected_exit_status7--Dproperty1=test---bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [ 90%]
tests/test_openjdk.py::test_jdk_extended[local-system_env-expected_strings8-expected_exit_status8---ENV1=test-bci/openjdk:11 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:11] PASSED [ 95%]
tests/test_openjdk.py::test_jdk_extended[local-system_env-expected_strings8-expected_exit_status8---ENV1=test-bci/openjdk:17 from registry.opensuse.org/devel/bci/sle-15-sp4/containerfile/bci/openjdk:17] PASSED [100%]
```

So issue is resolved, this is already the case

Reply:
I follow @dcermak recommendation here.